src/dataflow/src/decode/mod.rs

```diff
@@ -366,72 +365,69 @@ where
 
     let mut value_decoder = get_decoder(value_encoding, debug_name, operators, true, metrics);
 
-    // The Debezium deduplication and upsert logic rely on elements for the same key going to the same worker.
-    // Other decoders don't care; so we distribute things round-robin (i.e., by "position"), and
-    // fall back to arbitrarily hashing by value if the upstream didn't give us a position.
-    // TODO(petrosagg): this function should accept a pact from the caller
-    let use_key_contract = matches!(envelope, SourceEnvelope::Debezium(..));
-    let results = stream.unary_frontier(
-        Exchange::new(move |x: &SourceOutput<Option<Vec<u8>>, Option<Vec<u8>>>| {
-            if use_key_contract {
-                x.key.hashed()
-            } else if let Some(position) = x.position {
+    let dist: fn(&SourceOutput<Option<Vec<u8>>, Option<Vec<u8>>>) -> _ = match envelope {
+        SourceEnvelope::Debezium(dbz) => match dbz.mode {
```

Comment:
We should add a comment here, something like:

> NOTE: ON PAIN OF DEATH, you must keep this exchange function in sync with the one that we use in `debezium::whatnot::render()`, bla bla, because otherwise bla bla bad things will happen.

Reply:
Ah true, I still had the older version of the PR cached in my head... :see_no_evil: 

I feel a bit nervous that the other piece of code in `render` relies on this being done correctly, but the tests would at least catch it.